python/modules/IcePy/Types.cpp

```diff
@@ -3809,11 +3805,28 @@ IcePy::ExceptionWriter::ExceptionWriter(const PyObjectHandle& ex, const Exceptio
     }
 }
 
-IcePy::ExceptionWriter::~ExceptionWriter()
+IcePy::ExceptionWriter::ExceptionWriter(const ExceptionWriter& other)
 {
-    AdoptThread adoptThread; // Ensure the current thread is able to call into Python.
+    // Exception writer copy constructor can be called from a C++ thread when IcePy throws a ExceptionWriter object
```

Comment:
throws a_n_ ExceptionWriter object.

It's not immediately clear to me how this happens.  If you search the code, the 2 throw ExceptionWriter are already within an AdoptThread scope.

And I suspect the ExceptionWriter constructed in Operation.cpp is too.

Reply:
See https://github.com/zeroc-ice/ice/issues/1955 C++ calls the exception copy constructor.

MSVC calls it from current_exception, see also https://quuxplusone.github.io/blog/2019/05/11/msvc-what-are-you-doing/